File: cli_based_assistant.py
import argparse
import boto3
from botocore.exceptions import ClientError

def ask_claude(user_message, client, model_id):
    # Start a conversation with the user message.
    conversation = [
        {
            "role": "user",
            "content": [{"text": user_message}],
        }
    ]

    try:
        # Send the message to the model, using a basic inference configuration.
        response = client.converse(
            modelId=model_id,
            messages=conversation,
            inferenceConfig={"maxTokens":1000, "temperature":1},
            additionalModelRequestFields={"top_k":250}
        )

        # Extract and return the response text.
        response_text = response["output"]["message"]["content"][0]["text"]
        return response_text

    except (ClientError, Exception) as e:
        return f"ERROR: Can't invoke '{model_id}'. Reason: {e}"

def main():
    # Create a Bedrock Runtime client in the AWS Region you want to use.
    client = boto3.client("bedrock-runtime", region_name="eu-west-2")
    model_id = "anthropic.claude-3-sonnet-20240229-v1:0"

    print("Welcome to CMND.ai Claude 3 Sonnet Assistant")
    while True:
        user_message = input("User: ")
        if user_message.lower() in ["exit", "quit"]:
            print("Claude: Goodbye!")
            break
        
        response = ask_claude(user_message, client, model_id)
        print(f"Claude: {response}")

if __name__ == "__main__":
    main()

File: cli_based_assistant.js
const {
  BedrockRuntimeClient,
  InvokeModelCommand,
} = require("@aws-sdk/client-bedrock-runtime");

async function askClaude(userMessage, client, modelId) {
  // Start a conversation with the user message.
  const conversation = [
    {
      role: "user",
      content: [{ type: "text", text: userMessage }],
    },
  ];

  try {
    // Prepare the request body
    const body = JSON.stringify({
      anthropic_version: "bedrock-2023-05-31",
      max_tokens: 1000,
      messages: conversation,
      temperature: 1,
      top_k: 250,
    });

    // Send the message to the model, using a basic inference configuration.
    const command = new InvokeModelCommand({
      modelId: modelId,
      body: body,
      contentType: "application/json",
      accept: "application/json",
    });

    const response = await client.send(command);

    // Parse the response body
    const responseBody = JSON.parse(new TextDecoder().decode(response.body));

    // Extract and return the response text.
    const responseText = responseBody.content[0].text;
    return responseText;
  } catch (error) {
    return `ERROR: Can't invoke '${modelId}'. Reason: ${error}`;
  }
}

async function main() {
  // Create a Bedrock Runtime client in the AWS Region you want to use.
  const client = new BedrockRuntimeClient({ region: "eu-west-2" });
  const modelId = "anthropic.claude-3-sonnet-20240229-v1:0";

  console.log("Welcome to CMND.ai Claude Sonnet 3.5 Assistant");

  const readline = require("readline").createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  const askQuestion = (query) =>
    new Promise((resolve) => readline.question(query, resolve));

  while (true) {
    const userMessage = await askQuestion("User: ");
    if (
      userMessage.toLowerCase() === "exit" ||
      userMessage.toLowerCase() === "quit"
    ) {
      console.log("Claude: Goodbye!");
      readline.close();
      break;
    }

    const response = await askClaude(userMessage, client, modelId);
    console.log(`Claude: ${response}`);
  }
}

main();
